### PR TITLE
FC-1071 Integer arguments come in as strings

### PR DESCRIPTION
**Pull request by bplatz**
Attempts to parse all command line arguments into integers if possible.

### CODE REVIEW

**Review by jdorety — APPROVED** [review]

File: src/fluree/util.clj [hunk 1]
```diff
@@ -112,8 +112,10 @@
                                       :else
                                       (try (Integer/parseInt v) ;; parse to integer if possible
                                            (catch Exception _ v)))]
-                             (assoc acc k* v*))) {} opts)]
-    [opts-map args*]))
+                             (assoc acc k* v*))) {} opts)
+        ;; parse any args possible into integers
+        args**   (map #(try (Integer/parseInt %) (catch Exception _ %)) args*)]
+    [opts-map args**]))
```
Comment (cap10morgan) [inline src/fluree/util.clj:118]: Do we want to try parsing all args as ints and keep any that parse (if I'm reading it correctly)? This seems like a great application of clojure.spec and input coercion.

File: src/fluree/ledger/core.clj [hunk 2]
```diff
@@ -112,6 +112,7 @@
 
 (defn read-block
   [data-dir ledger int]
+  (assert (int? int) (str "Read-block needs an integer block number, sent: " int " of type: " (type int)))
```
Comment (cap10morgan) [inline src/fluree/ledger/core.clj:115]: This would also be a good place for some clojure.spec love. :)

**Review by cap10morgan — APPROVED** [review]
Some questions about the aggressiveness of the approach and whether or not clojure.spec might be better. But no blockers. :)

File: src/fluree/util.clj (hunk 1 above)
Comment (bplatz) [reply src/fluree/util.clj:118]: There are only 4 possible args, and the only time an int will be used is to specify blocks... so this approach has no functional downside, no performance downside... only a visual downside. I'd rather write a function that used a regex and then more precisely parsed to integer (which is what parseInt does) than pull in a library dependency for just this purpose. I'm not sure I like the idea of spec being used for argument coercion (not sure I don't like it either, but that's not the design intention of clojure.spec so far as I know)... but for use in testing I'm a big fan.

**Pull request merged**